### ISSUE TITLE
大神接棒，YOLOv4来了！ - 知乎

### ISSUE THREAD

**Issue by madobet** [madobet/wwwclip#141]
> 编辑：Amusi  
> Date：2020-04-24  
> 来源：[CVer](https://link.zhihu.com/?target=https%3A//mp.weixin.qq.com/s/Ua3T-DOuzmLWuXfohEiVFw)微信公众号  
> 链接：[大神接棒，YOLOv4 来了！](https://link.zhihu.com/?target=https%3A//mp.weixin.qq.com/s/Ua3T-DOuzmLWuXfohEiVFw)

## **前言**

今天刷屏的动态一定是 YOLOv4！

![](https://pic3.zhimg.com/v2-5f48982c03cdc7eb1a1324c6ee84f32a_b.jpg)

本文 Amusi 会跟大家说一下在别处看不到内容（大神接棒），欢迎继续阅读！

之前，YOLO 系列 (v1-v3) 作者 Joe Redmon 宣布不再继续 CV 方向的研究，引起学术圈一篇哗然。[YOLO 之父宣布退出 CV 界，坦言无法忽视自己工作带来的负面影响](https://link.zhihu.com/?target=http%3A//mp.weixin.qq.com/s%3F__biz%3DMzUxNjcxMjQxNg%3D%3D%26mid%3D2247496822%26idx%3D2%26sn%3D89fe4610a3993204c5c21058d9923c5d%26chksm%3Df9a186f9ced60fefd543f86eaa05125b185560e0c6f27598575eaac5045c1b421b86b13421ea%26scene%3D21%23wechat_redirect)

![](https://pic2.zhimg.com/v2-2f50494720dcee903cb294833d8b2b79_b.jpg)

_推文链接：[https://twitter.com/jeremyphoward/status/1230610470991589376](https://link.zhihu.com/?target=https%3A//twitter.com/jeremyphoward/status/1230610470991589376)_

Amusi 特意去谷歌学术上搜索了一下 YOLOv1-v3 的引用量，累计破 16000+！

![](https://pic2.zhimg.com/v2-14295944fceba8f781a200627f01e7fd_b.jpg)

## **大神接棒，YOLOv4 来了！**

当大家以为再也见不到 YOLOv4 的时候，然鹅今天 **YOLOv4** 来了！

YOLOv4 的作者阵容里并没有 Joe Redmon，也验证了大神曾说不再继续 CV 研究的这条消息。**但都木有 YOLO 之父 Joe Redmon 的论文，其名字为什么还敢叫 YOLOv4 呢，不怕被喷么？**

这里 Amusi 花点时间跟大家介绍一下这个有趣的事情。

先说说：**大神接棒的事情**

停更两年之久的的 YOLO github 官网正式更新 README，那么更新了什么呢？接着往下看

YOLO 官网：[https://github.com/pjreddie/darknet](https://link.zhihu.com/?target=https%3A//github.com/pjreddie/darknet)

![](https://pic4.zhimg.com/v2-8c347cf315f73aa2922b7081973e3883_b.jpg)

YOLO 官方 github 正式加入 YOLOv4 的论文和代码链接，也意味着 YOLOv4 得到了 Joe Redmon 的认可，也代表着 YOLO 的**停更与交棒**。

![](https://pic4.zhimg.com/v2-5a381e10bc63f7d1aab38749cdcd8eff_b.jpg)

关于 YOLOv4，Amusi 曾写过这样一篇文章来**预宣传**（现在想想，应该是全网第一个这样介绍的）：[等待 YOLOv4 的期间，它还在更新](https://zhuanlan.zhihu.com/p/97382064)

YOLOv4 的一作是 Alexey Bochkovskiy，用过 YOLO 的同学，特别是用过 Windows 版 YOLO 的同学对这个名字一定很熟悉！

因为他就是 darknet 另一个 github 版本的维护者，也就是 YOLOv4 的代码库：

[https://github.com/AlexeyAB/darknet](https://link.zhihu.com/?target=https%3A//github.com/AlexeyAB/darknet)

据 Amusi 猜测推断，Alexey Bochkovskiy 是一名独立研究员，这里更愿意称为之 YOLO 接棒者，YOLO 社区推动者。这里对 Alexey 不过多赘述，想了解的同学可以看一下：[等待 YOLOv4 的期间，它还在更新](https://link.zhihu.com/?target=http%3A//mp.weixin.qq.com/s%3F__biz%3DMzUxNjcxMjQxNg%3D%3D%26mid%3D2247494523%26idx%3D2%26sn%3De6af5d9a75d9d4c1fac342ab4ee71928%26chksm%3Df9a199f4ced610e2e04d1eb24d1a8151e363f1437a836cd26617aab9832616662280593c890a%26scene%3D21%23wechat_redirect)

值得一提的是，这个版本的 darknet 的提交数已经来到 1777 次。试想一下，你一天提交一次更新，那么就需要你连续近 5 年不停更！

我们一般只会用两个字来形容这种人：**码怪**！

![](https://pic2.zhimg.com/v2-ee138d7e2760d1c06e456849b75dea31_b.jpg)

## **YOLOv4 正文**

![](https://pic1.zhimg.com/v2-a1a8209035bce3342d7d1d08e49bb14c_b.jpg)

YOLOv4: Optimal Speed and Accuracy of Object Detection

论文：[https://arxiv.org/abs/2004.10934](https://link.zhihu.com/?target=https%3A//arxiv.org/abs/2004.10934)

代码：[https://github.com/AlexeyAB/darknet](https://link.zhihu.com/?target=https%3A//github.com/AlexeyAB/darknet)

大家一定被文章开头的图片吸引了，位于图中靠右上角的 YOLOv4 多么 "亮眼"，越靠右上角意味着 AP 越高、速度 FPS 越快！而且 YOLO 被大家追捧的原因之一就是：快而准。

![](https://pic3.zhimg.com/v2-5f48982c03cdc7eb1a1324c6ee84f32a_b.jpg)

YOLOv4 在 COCO 上，可达 43.5％ AP，速度高达 65 FPS！

YOLOv4 的特点是集大成者，俗称堆料。但最终达到这么高的性能，一定是不断尝试、不断堆料、不断调参的结果，给作者点赞。下面看看堆了哪些料：

-   **Weighted-Residual-Connections (WRC)**
-   **Cross-Stage-Partial-connections (CSP)**
-   **Cross mini-Batch Normalization (CmBN)**
-   **Self-adversarial-training (SAT)**
-   **Mish-activation**
-   **Mosaic data augmentation**
-   **CmBN**
-   **DropBlock regularization**
-   **CIoU loss**

本文的主要贡献如下：

1.  提出了一种高效而强大的目标检测模型。它使每个人都可以使用 1080 Ti 或 2080 Ti GPU 训练超快速和准确的目标检测器（牛逼！）。

2.  在检测器训练期间，验证了 SOTA 的 Bag-of Freebies 和 Bag-of-Specials 方法的影响。

3.  改进了 SOTA 的方法，使它们更有效，更适合单 GPU 训练，包括 CBN \[89]，PAN \[49]，SAM \[85] 等。文章将目前主流的目标检测器框架进行拆分：input、backbone、neck 和 head 具体如下图所示：

![](https://pic3.zhimg.com/v2-88cca6b8075ac239bc2934b4e980e066_b.jpg)

-   对于 GPU，作者在卷积层中使用：CSPResNeXt50 / CSPDarknet53  
-   对于 VPU，作者使用分组卷积，但避免使用（SE）块 - 具体来说，它包括以下模型：EfficientNet-lite / MixNet / GhostNet / MobileNetV3

作者的目标是在输入网络分辨率，卷积层数，参数数量和层输出（filters）的数量之间找到最佳平衡。文中称：CSPDarknet53 在检测上的表现要优于 CSPResNext50，关于 CSP，不了解的同学可以看一下这篇文章：

[增强 CNN 学习能力的 Backbone:CSPNet](https://link.zhihu.com/?target=http%3A//mp.weixin.qq.com/s%3F__biz%3DMzUxNjcxMjQxNg%3D%3D%26mid%3D2247497796%26idx%3D3%26sn%3D6e009c4bdb960322507fccff1ad45cb6%26chksm%3Df9a18acbced603dd26f46878077de11a937fa2a6405e99d03bfd9379c91d53959cd94e980f05%26scene%3D21%23wechat_redirect)

![](https://pic2.zhimg.com/v2-b3fb29befb7cca99cad75f917157fa19_b.jpg)

总结一下 YOLOv4 框架：

-   **Backbone：CSPDarknet53**
-   **Neck：SPP，PAN**
-   **Head：YOLOv3**

**YOLOv4 =** **CSPDarknet53+SPP+PAN+YOLOv3**

其中 YOLOv4 用到相当多的技巧：

-   **用于 backbone 的 BoF：CutMix 和 Mosaic 数据增强，DropBlock 正则化，Class label smoothing**
-   **用于 backbone 的 BoS：Mish 激活函数，CSP，MiWRC**
-   **用于检测器的 BoF：CIoU-loss，CmBN，DropBlock 正则化，Mosaic 数据增强，Self-Adversarial 训练，消除网格敏感性，对单个 ground-truth 使用多个 anchor，Cosine annealing scheduler，最佳超参数，Random training shapes**
-   **用于检测器的 Bos：Mish 激活函数，SPP，SAM，PAN，DIoU-NMS**

看看 YOLOv4 部分组件：

![](https://pic1.zhimg.com/v2-e7f102f8c2e4a09d904922306a3eb960_b.jpg)
![](https://pic1.zhimg.com/v2-418093ab12a48f9dd916d8315093ce88_b.jpg)

感受一下 YOLOv4 实验的充分性（调参的艺术）

![](https://pic3.zhimg.com/v2-ff0e5039573acb57e40ee4c4e95d47e6_b.jpg)
![](https://pic1.zhimg.com/v2-665c30a42709cda5092c376615b10ec8_b.jpg)

感受一下性能炸裂的 YOLOv4 实验结果

![](https://pic1.zhimg.com/v2-3c5f16ecee9dad601821ecf9c13c01e0_b.jpg)
![](https://pic2.zhimg.com/v2-b70240ec3b4de4b950b007d7492909e5_b.jpg)
![](https://pic3.zhimg.com/v2-a7934c75535ddb71a6901844bf0f2986_b.jpg)
![](https://pic1.zhimg.com/v2-ad05acd5fa8fc093f70111df3ebea770_b.jpg)

论文篇幅很长，高达 17 页，干货满满！推荐大家阅读原文和源码进行深入理解。

欢迎各位 CVers 点赞支持！也推荐大家关注 [计算机视觉论文速递](https://zhuanlan.zhihu.com/c_172507674) 知乎专栏和 [CVer](https://link.zhihu.com/?target=https%3A//mp.weixin.qq.com/s/moRNkCaBDtYZgXwaZDXBwA) 微信公众号，可以快速了解到最新优质的 CV 论文。

**跟我一起喊：大神接棒，YOLOv4 来了！**

## **推荐阅读**

[ResNet 最强改进版来了！ResNeSt：Split-Attention Networks](https://zhuanlan.zhihu.com/p/132655457)

[一文看尽 10 篇目标检测最新论文（MetaOD/P-RSDet/MatrixNets 等）](https://zhuanlan.zhihu.com/p/107942332)

[一文看尽 9 篇语义分割最新论文（GPSNet/Graph-FCN/HMANet 等）](https://zhuanlan.zhihu.com/p/108274663)

[周志华教授：如何做研究与写论文](https://zhuanlan.zhihu.com/p/98747105)

[等待 YOLOv4 的期间，它还在更新](https://zhuanlan.zhihu.com/p/97382064)

[大盘点 | 2019 年 4 篇目标检测算法最佳综述](https://zhuanlan.zhihu.com/p/94090477)

 <https://zhuanlan.zhihu.com/p/135909702>
